tests/NServiceKit.Text.Tests/Utils/DateTimeSerializerTests.cs

```diff
@@ -371,6 +371,52 @@ public void DateTime_Is_Serialized_As_Utc_and_Deserialized_as_local()
             {
                 Assert.AreEqual(DateTimeKind.Utc, TypeSerializer.DeserializeFromString<TestObject>(TypeSerializer.SerializeToString<TestObject>(testObject)).Date.Kind);
             }
-        }        
+        }
+
+        /// <summary>
+        /// DateTime with DateTimeKind.Unspecified should be treated as UTC (rather than Local) when AssumeUtc is true
+        /// </summary>
+        [Test]
+        public void DateTime_Unspecified_Is_Serialized_And_Deserialized_As_Utc_When_AssumeUtc_Flag_Is_True()
+        {
+            var dateTimeUnspecified = new DateTime(2013, 1, 1, 0, 0, 1, DateTimeKind.Unspecified);
+
+            // Using JsonSerializer becayse TypeSerializer doesn't call DateTimeSerializer.WriteWcfJsonDate
+            var deserialized = JsonSerializer.DeserializeFromString<DateTime>(JsonSerializer.SerializeToString<DateTime>(dateTimeUnspecified));
```

Comment:
Had to use JsonSerializer here because it seems TypeSerializer is only calling the DateTimeSerializer methods on deserialize... not on serialize. JsonSerializer used it for both.